### PR TITLE
raft of minor cleanups

### PR DESCRIPTION
**Pull request by gavinking**
- fix some warnings from IntelliJ
- JavaDoc -> Javadoc
- remove some more missed occurrences of "slice"

### CODE REVIEW

File: api/src/main/java/jakarta/data/repository/Query.java [hunk 1]
```diff
@@ -114,8 +114,8 @@
      *
      * <p>The default value of empty string indicates that no counting query
      * is provided. A counting query is unnecessary when pagination is
-     * performed with slices instead of pages and when pagination is
-     * not used at all.</p>
+     * performed with {@link jakarta.data.page.PageRequest#withTotal()} or
+     * when pagination is not used at all.</p>
```
Comment (njr-11) [inline api/src/main/java/jakarta/data/repository/Query.java:118]:
```suggestion
     * performed with {@link jakarta.data.page.PageRequest#withoutTotal()} or
     * when pagination is not used at all.</p>
```

**Review by njr-11 — COMMENTED** [review]
One typo to correct, otherwise looks good.

File: api/src/main/java/jakarta/data/repository/Query.java (hunk 1 above)
Comment (gavinking) [reply api/src/main/java/jakarta/data/repository/Query.java]: I could _swear_ that this is what I typed!

**Review by njr-11 — APPROVED** [review]

**Comment by otaviojava** [comment]
@gavinking could you fix this conflict?

**Comment by gavinking** [comment]
> @gavinking could you fix this conflict?

Done.

**Pull request merged**